src/ssh/cockpitsshrelay.c

```diff
@@ -2242,9 +2234,12 @@ cockpit_ssh_relay_start (CockpitSshRelay *self,
   in = dup (0);
   g_assert (in >= 0);
 
+  out = dup (0);
+  g_assert (out >= 0);
```

Comment:
Why is that necessary? Can't in-fd and out-fd both use the same fd? (and `in` would then be renamed to `inout` or maybe `bridge_fd`)

Reply:
CockpitPipe takes ownership and wants to close each of them independently.